src/Data/Codec/JSON/Record.purs

```diff
@@ -53,39 +82,80 @@ optional = Optional
 -- | The class used to enable the building of `Record` codecs by providing a
 -- | record of codecs.
 class RowListCodec (rl ∷ RL.RowList Type) (ri ∷ Row Type) (ro ∷ Row Type) | rl → ri ro where
-  rowListCodec ∷ ∀ proxy. proxy rl → Record ri → CJ.PropCodec (Record ro)
+  rowListCodec ∷ Record ri → CJ.PropCodec (Record ro)
 
-instance rowListCodecNil ∷ RowListCodec RL.Nil () () where
-  rowListCodec _ _ = CJ.record
+instance RowListCodec RL.Nil () () where
+  rowListCodec _ = CJ.record
 
-instance rowListCodecConsOptional ∷
+instance
   ( RowListCodec rs ri' ro'
   , R.Cons sym (Optional a) ri' ri
   , R.Cons sym (Maybe a) ro' ro
   , IsSymbol sym
   ) ⇒
   RowListCodec (RL.Cons sym (Optional a) rs) ri ro where
-  rowListCodec _ codecs =
+  rowListCodec codecs =
     CJ.recordPropOptional (Proxy ∷ Proxy sym) codec tail
     where
     codec ∷ CJ.Codec a
     codec = coerce (Rec.get (Proxy ∷ Proxy sym) codecs ∷ Optional a)
 
     tail ∷ CJ.PropCodec (Record ro')
-    tail = rowListCodec (Proxy ∷ Proxy rs) ((unsafeCoerce ∷ Record ri → Record ri') codecs)
+    tail = rowListCodec @rs ((unsafeCoerce ∷ Record ri → Record ri') codecs)
```

Comment:
Introducing type applications raises the minimum compiler version to run this to 0.15.x - I'd say we should either not use them at all or do a proper breaking change where all the `Proxy` are turned into type applications.

Reply:
I'd be fine converting everything to use type applications, so could accept this as-is and then update the rest in a follow up PR.

Reply:
Well, I don't have a problem either way. And since this is your repo @garyb, I'll do whatever you say.